### PR TITLE
feat: view and filter ideas at homepage

### PR DESCRIPTION
**Pull request by letuan32**
Base view, search, filter, and sort ideas on the homepage.
Will implement other features in next commits

### CODE REVIEW

File: COMP1640/ViewModels/Catalog/GetIdeaIndexRequest.cs [hunk 1]
```diff
@@ -0,0 +1,12 @@
+namespace COMP1640.ViewModels.Common;
```
Comment (HoangMinhTrong) [inline COMP1640/ViewModels/Catalog/GetIdeaIndexRequest.cs]: In my case, I think you need to split in folder Requests

File: COMP1640/ViewModels/Catalog/IdeaIndexResponse.cs [hunk 2]
```diff
@@ -0,0 +1,22 @@
+using Microsoft.AspNetCore.Mvc.Rendering;
```
Comment (HoangMinhTrong) [inline COMP1640/ViewModels/Catalog/IdeaIndexResponse.cs]: In my case, I think you need to split in folder Responses

File: COMP1640/ViewModels/Catalog/IdeaIndexItem.cs [hunk 3]
```diff
@@ -0,0 +1,56 @@
+using System.Linq.Expressions;
+using Domain;
+
+namespace COMP1640.ViewModels.Common;
```
Comment (HoangMinhTrong) [inline COMP1640/ViewModels/Catalog/IdeaIndexItem.cs]: In my case, I think you need to split in folder Responses

File: COMP1640/Controllers/HomeController.cs [hunk 4]
```diff
@@ -1,19 +1,62 @@
...
-            return View();
+            var ideas = await _ideaService.GetIdeaIndexAsync(request);
+            
+            var response = new IdeaIndexResponse()
+            {
+                IdeaIndexItems = ideas,
+                Categories = await _categoryService.GetCategoryPicklistAsync(),
+                SortOptionPicklist = IdeaIndexOption.SortOptionPicklist,
+                
+                // Comment sort option switch since just sort by descending
```
Comment (HoangMinhTrong) [inline COMP1640/Controllers/HomeController.cs]: What purpose of this comment? if this not have any value, please remove it

**Review by HoangMinhTrong — COMMENTED** [review]
Reviewed, please check it and fix

**Comment by letuan32** [comment]
Resolved. Please review again

**Pull request merged**